core/alumet/src/units.rs

```diff
@@ -238,6 +238,30 @@ impl Display for PrefixedUnit {
     }
 }
 
+impl FromStr for PrefixedUnit {
+    type Err = anyhow::Error;
```

Comment:
Let's create a dedicated error type. You can use [`thiserror`](https://crates.io/crates/thiserror) to make it easy.